src/main/java/org/jenkinsci/plugins/scriptsecurity/casc/ScriptApprovalConfigurator.java

```diff
@@ -0,0 +1,77 @@
+package org.jenkinsci.plugins.scriptsecurity.casc;
+
+import hudson.Extension;
+import io.jenkins.plugins.casc.Attribute;
+import io.jenkins.plugins.casc.BaseConfigurator;
+import io.jenkins.plugins.casc.ConfigurationContext;
+import io.jenkins.plugins.casc.RootElementConfigurator;
+import io.jenkins.plugins.casc.impl.attributes.MultivaluedAttribute;
+import io.jenkins.plugins.casc.model.CNode;
+import io.jenkins.plugins.casc.model.Mapping;
+import org.jenkinsci.plugins.scriptsecurity.scripts.ScriptApproval;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.CheckForNull;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @author <a href="mailto:ohad.david@gmail.com">Ohad David</a>
+ */
+@Extension(optional = true)
+@Restricted(NoExternalUse.class)
+public class ScriptApprovalConfigurator extends BaseConfigurator<ScriptApproval> implements RootElementConfigurator<ScriptApproval> {
+
+
+    @Override
+    public String getName() {
+        return "scriptApproval";
+    }
+
+    @Override
+    public Class<ScriptApproval> getTarget() {
+        return ScriptApproval.class;
+    }
+
+    @Override
+    public Set<Attribute<ScriptApproval, ?>> describe() {
+        final Set<Attribute<ScriptApproval, ?>> describe = new HashSet<>();
+        describe.add(new MultivaluedAttribute<ScriptApproval, String>("approvedSignatures", String.class)
+                .getter((ScriptApproval target) -> Arrays.asList(target.getApprovedSignatures()))
+                .setter((target, value) -> value.forEach(signature -> uncheckedApproveSignature(target, signature))));
+        return describe;
+    }
+
+    private static String[][] uncheckedApproveSignature(ScriptApproval target, String signature) {
+        try {
+            return target.approveSignature(signature);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
```

Comment:
This is a bad idea, it will still make JCasC to fail in unpredictable way.
I would rather expect an explicit failure to happen. If it is not possible, better to log a warning to the system log

Reply:
@oleg-nenashev - Resolved both comments.

Reply:
The current code still throws unchecked exceptions. CC @ndeloof who can comment about the viability of such approach

Reply:
I personally dislike checked exception, they prevented me to use method references in streams few times, and I use to just forward to caller, which at the end dump some stacktrace whatever happens.  I'd prefer java only define unchecked ones.

Reply:
There is Kotlin for it 🤷‍♂️

Reply:
yes indeed. Maybe this is what we need to "Jenkins Bolt" to actually renew Jenkins developer experience